tests/unit/components/sl-date-range-picker-test.js

```diff
@@ -0,0 +1,352 @@
+        '01/01/2015'
+    );
+});
+
+test( 'Events from start date input are removed upon willClearRender', function( assert ) {
+    let component = this.subject();
+    let startDateInput = this.$( '.sl-daterange-start-date input' )[0];
+
+    assert.equal(
+        Ember.typeOf( $._data( startDateInput, 'events' ).change ),
+        'array',
+        'Start date input has change event listener after render'
+    );
+
+    component.trigger( 'willClearRender' );
+
+    assert.strictEqual(
+        $._data( startDateInput, 'events' ),
+        undefined,
+        'Start date input has no event listeners after willClearRender'
+    );
+});
+
+test( 'label, startDatePlaceholder, and endDatePlaceholder are undefined by default', function( assert ) {
+    let component = this.subject();
+
+    assert.strictEqual(
+        component.get( 'label' ),
+        undefined,
+        'label is undefined by default'
+    );
+
+    assert.strictEqual(
+        this.$( 'label' )[0],
+        undefined,
+        'No label is created when label is undefined'
+    );
+
+    assert.strictEqual(
+        component.get( 'startDatePlaceholder' ),
+        undefined,
+        'startDatePlaceholder is undefined by default'
+    );
+
+    assert.strictEqual(
+        this.$( '.sl-daterange-start-date input' ).prop( 'placeholder' ),
+        '',
+        'Start date input placeholder is empty when startDatePlaceholder is undefined'
+    );
+
+    assert.strictEqual(
+        component.get( 'endDatePlaceholder' ),
+        undefined,
+        'endDatePlaceholder is undefined by default'
+    );
+
+    assert.strictEqual(
+        this.$( '.sl-daterange-end-date input' ).prop( 'placeholder' ),
+        '',
+        'End date input placeholder is empty when endDatePlaceholder is undefined'
+    );
+});
+
+test( 'label is accepted as a parameter', function( assert ) {
+    let labelText = 'lorem ipsum';
+    let component = this.subject({ label: labelText });
+
+    assert.equal(
+        component.get( 'label' ),
+        labelText,
+        'label was accepted as a parameter'
+    );
+
+    assert.equal(
+        this.$( 'label' ).html(),
+        labelText,
+        'label element was created with label parameter text'
+    );
+
+    assert.equal(
+        this.$( 'label' ).prop( 'for' ),
+        component.get( 'inputElementId' ),
+        'label element has the correct for property'
+    );
+
+    assert.strictEqual(
+        component.get( 'startDatePlaceholder' ),
+        undefined,
+        'startDatePlaceholder is undefined by default'
+    );
+
+    assert.strictEqual(
+        this.$( '.sl-daterange-start-date input' ).prop( 'placeholder' ),
+        '',
+        'Start date input placeholder is empty when startDatePlaceholder is undefined'
+    );
+
+    assert.strictEqual(
+        component.get( 'endDatePlaceholder' ),
+        undefined,
+        'endDatePlaceholder is undefined by default'
+    );
+
+    assert.strictEqual(

```

Comment:
By testing each individual property behavior, as appropriate, individually and then the methods and logic that interact with them, as a whole this module will test the scenario you describe.  The non-technical aspect of this approach is where do you draw the line?  Do you put every property in here?  What about those that are dependent on others?  Where do you stop and and what point are all the things you want to test just being done in a single test.  I understand what you are saying but do not agree that it's the best approach, as explaining in my opening sentence.